api/nakadi-event-bus-api.yaml

```diff
@@ -606,22 +488,388 @@ paths:
           description: Partition id
           type: string
           required: true
-        - name: X-Flow-Id
+        - $ref: '#/parameters/XFlowId'
+      responses:
+        '200':
+          description: OK
+          schema:
+            $ref: '#/definitions/Partition'
+        '401':
+          description: Client is not authenticated
+          schema:
+            $ref: '#/definitions/Problem'
+
+  /subscriptions:
+    post:
+      tags:
+        - managed-api
+      security:
+        - oauth2: ['nakadi.event_stream.read']
+      description: |
+        This endpoint creates a subscription for EventTypes. The subscription is needed to be able to
+        consume events from EventTypes in a high level way when Nakadi stores the offsets and manages the
+        rebalancing of consuming clients.
+        The subscription is identified by its key parameters (owning_application, event_types, consumer_group). If
+        this endpoint is invoked several times with the same key subscription properties in body - the
+        subscription will be created only once and for all other calls it will just return the subscription
+        that was already created.
+      parameters:
+        - name: subscription
+          in: body
+          description: Subscription to create
+          schema:
+            $ref: '#/definitions/Subscription'
+          required: true
+      responses:
+        '200':
+          description: |
+            Subscription for such parameters already exists. Returns subscription object that already
+            existed.
+          schema:
+            $ref: '#/definitions/Subscription'
+        '201':
+          description: Subscription was successfuly created. Returns subscription object that was created.
+          schema:
+            $ref: '#/definitions/Subscription'
+        '401':
+          description: Client is not authenticated
+          schema:
+            $ref: '#/definitions/Problem'
+        '403':
+          description: Client is not authorized to perform this operation
+          schema:
+            $ref: '#/definitions/Problem'
+        '400':
+          description: Bad Request
+          schema:
+            $ref: '#/definitions/Problem'
+        '422':
+          description: Unprocessable Entity
+          schema:
+            $ref: '#/definitions/Problem'
+        '500':
+          description: Server error
+          schema:
+            $ref: '#/definitions/Problem'
+        '503':
+          description: Service (temporarily) unavailable
+          schema:
+            $ref: '#/definitions/Problem'
+    get:
+      tags:
+        - managed-api
+      security:
+        - oauth2: ['nakadi.event_stream.read']
+      description: Lists all subscriptions that exist in a system.
```

Comment:
> If a client doesn't know that there will be a next-link (or what it means), it is not going to follow it, so those clients have no chance to get all the subscriptions.

This is correct. If a physical limit is put on the subscription array size, and it would be inadvisable not to do that, then those clients won't see all subs. I think paging needs to be part of this API, and it should not be based on arrays.